ui/src/features/project/roles/rule-editor.tsx

```diff
@@ -0,0 +1,119 @@
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Button, Select, SelectProps } from 'antd';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import { FieldContainer } from '@ui/features/common/form/field-container';
+import { MultiStringEditor } from '@ui/features/common/form/multi-string-editor';
+import { PolicyRule } from '@ui/gen/k8s.io/api/rbac/v1/generated_pb';
+
+const availableResources = [
+  // core
+  'events',
+  'secrets',
+  'serviceaccounts',
+  // rbac.authorization.k8s.io
+  'rolebindings',
+  'roles',
+  // kargo.akuity.io
+  'freights',
+  'stages',
+  'promotions',
+  'warehouses',
+  'secret',
+  // argoproj.io
+  'analysisruns',
+  'analysistemplates'
+];
+const availableVerbs = ['create', 'update', 'delete', 'patch', 'list', 'promote'];
```

Comment:
This seems to be missing `get`

Reply:
Can we also have `*` as a verb? You do not need to expand it on the front end. The back end already does.

Also... idk if this is too much to ask, but the promote verb is only applicable to Stages. Is it possible to only have that option if Stage is selected as the resource type?